aioshelly/block_device/coap.py

```diff
@@ -229,27 +234,30 @@ def subscribe_updates(
         self.subscriptions[ip_or_device_id] = message_received
         return lambda: self.subscriptions.pop(ip_or_device_id)
 
-    async def __aenter__(self) -> "COAP":
+    async def __aenter__(self) -> Self:
```

Comment:
ruff complains to use `-> Self`
PYI034 `__aenter__` methods in classes like `COAP` usually return `self` at runtime